pg_shard.c

```diff
@@ -625,8 +634,13 @@ PlanSequentialScan(Query *query, int cursorOptions, ParamListInfo boundParams)
 	enable_indexscan = false;
 	enable_bitmapscan = false;
 
-	queryCopy = copyObject(query);
-	sequentialScanPlan = standard_planner(queryCopy, cursorOptions, boundParams);
+	/*
+	 * Remove all quals from query before calling planner. Remote workers will
+	 * apply quals for us: RowAndColumnFilterQuery ensures all quals are pushed
+	 * down during remote executions. The mutator returns a copy of the query.
+	 */
+	modifiedQuery = (Query *) RemoveFromExpressionQualsMutator((Node *) query, NULL);
```

Comment:
The `FromExpr` is actually a tree, so recursing down it to set all `quals` to `NULL` is most appropriate. I believe that we shouldn't ever see anything other than a lone node as we're not supporting `JOINs` at the moment, but it seemed more "correct" to recurse along the entire tree.

I could set `quals` to `NULL` and just document with a `TODO` or something that we would need to change this piece of code if we ever push down `JOIN`s.…